src/Symfony/Component/Config/Definition/ArrayNode.php

```diff
@@ -227,7 +227,7 @@ protected function finalizeValue($value)
             }
 
             if ($child->isDeprecated()) {
-                @trigger_error($child->getDeprecationMessage($name, $this->getPath()), E_USER_DEPRECATED);
+                deprecated('', '', $child->getDeprecationMessage($name, $this->getPath()));
```

Comment:
What about reorder arguments to make package and version arguments are optional?

Reply:
See my answer above. I think it's critical to give the package and the version - these should not be optional. Otherwise, good luck fixing those deprecations. Providing dedicated arguments means deprecation authors won't forget about that.

Reply:
I totally agree with Nicolas here. This should be added by the maintainers 👍🏻